app/code/Magento/Elasticsearch/Model/DataProvider/Base/Suggestions.php

```diff
@@ -111,10 +127,13 @@ public function __construct(
         $this->config = $config;
         $this->searchIndexNameResolver = $searchIndexNameResolver;
         $this->storeManager = $storeManager;
-        $this->fieldProvider = $fieldProvider;
```

Comment:
Removing a constructor argument sadly is a backward compatibility breaking change.
The scenario that someone extends from the class, overrides the constructor, and calls the parent constructor with the ordered arguments that include the FieldProvider would break.
Because of that, the $fieldProvider argument has to remain, even if it is no longer used.